src/server/ua_services_nodemanagement.c

```diff
@@ -924,6 +924,80 @@ recursiveTypeCheckAddChildren(UA_Server *server, UA_Session *session,
     return UA_STATUSCODE_GOOD;
 }
 
+/* Check if we got a valid browse name for the new node.
+ * For object nodes the BrowseName may only be null if the parent type has a
+ * 'DefaultInstanceBrowseName' property.
+ * */
+static UA_StatusCode
+checkValidBrowseName(UA_Server *server, UA_Session *session,
+							  const UA_Node **nodeptr, const UA_Node *type) {
+
+	UA_assert(type != NULL);
+	UA_StatusCode retval = UA_STATUSCODE_GOOD;
+	const UA_Node *node = *nodeptr;
+
+	if (node->nodeClass != UA_NODECLASS_OBJECT) {
+		if (UA_QualifiedName_isNull(&node->browseName))
```

Comment:
Check for UA_QualifiedName_isNull(&node->browseName) first and then for the nodeclass.
This removes the nested if.